### PR TITLE
New bench kate

### PR DESCRIPTION
**Pull request by kroos47**
Benchmarking credits to  anjan

### CODE REVIEW

File: kate/src/com.rs [hunk 1]
```diff
@@ -383,20 +417,21 @@ pub fn build_commitments(
...
-	// construct commitments in parallel
-	if block_dims.cols > MAX_BLOCK_COLUMNS as usize {
-		log::error!(
-			target: LOG_TARGET,
-			"Error on Block dimension {:?}",
-			block_dims
-		);
-	}
-	let public_params = testnet::public_params(MAX_BLOCK_COLUMNS as usize);
+	// if block_dims.cols > MAX_BLOCK_COLUMNS as usize {
```
Comment (aterentic-ethernal) [inline kate/src/com.rs:425]: Is this a reminder or leftover which could be removed?

File: kate/benches/kzg.rs [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+			dims.extend(&variate_rc(r, c));
+			c <<= 1;
+		}
+		r <<= 1;
+	}
+
+	dims.into_iter().unique().collect::<Vec<(usize, usize)>>()
+}
+
+// // Commitment builder routine candidate 1
```
Comment (aterentic-ethernal) [inline kate/benches/kzg.rs:59]: Do we need commented code?

File: kate/src/com.rs [hunk 3]
```diff
@@ -5,10 +5,10 @@ use std::{
 };
 
 use codec::Encode;
+// use bls12_381::{ G1Projective, Scalar};
```
Comment (aterentic-ethernal) [inline kate/src/com.rs:8]: More comments :)

File: kate/src/com.rs [hunk 4]
```diff
@@ -264,6 +267,48 @@ pub fn extend_data_matrix(
 	Ok(chunk_elements)
 }
 
+#[cfg(feature = "alloc")]
+pub fn par_extend_data_matrix(
```
Comment (aterentic-ethernal) [inline kate/src/com.rs:271]: Do we want to keep old `extend_data_matrix`?

File: kate/src/com.rs (hunk 1 above)
Comment (kroos47) [reply kate/src/com.rs:425]: yeh it can be removed

File: kate/src/com.rs (hunk 4 above)
Comment (kroos47) [reply kate/src/com.rs:271]: I don't think we need, we can go with parallel one!

File: kate/src/com.rs [hunk 5]
```diff
@@ -63,16 +64,18 @@ fn app_extrinsics_group_by_app_id(extrinsics: &[AppExtrinsic]) -> Vec<(u32, Vec<
...
+				if curr.0 == e.app_id {
+					curr.1.push(e.data.clone());
+				} else {
+					acc.push((e.app_id, vec![e.data.clone()]));
+				}
 			}
-		}
+		};
 		acc
 	})
```
Comment (aterentic-ethernal) [inline kate/src/com.rs:76]: Maybe nicer version:
```
	extrinsics.iter().fold(vec![], |mut acc, e| {
		match acc.last_mut() {
			Some((0, data)) => data.push(e.data.clone()),
			None | Some(_) => acc.push((e.app_id, vec![e.data.clone()])),
		}
		acc
	})
```

File: kate/src/com.rs [hunk 6]
```diff
@@ -59,20 +60,15 @@ const PADDING_TAIL_VALUE: u8 = 0x80;
...
-		} else {
-			let curr = acc.last_mut().unwrap();
-			if curr.0 == e.app_id {
-				curr.1.push(e.data.clone());
-			} else {
-				acc.push((e.app_id, vec![e.data.clone()]));
-			}
+	extrinsics.iter().fold(vec![], |mut acc, e| {
+		match acc.last_mut() {
+			Some((0, data)) => data.push(e.data.clone()),
```
Comment (aterentic-ethernal) [inline kate/src/com.rs:65]: Nicer but wrong, sorry 😆 

Correct version is:
```
			Some((app_id, data)) if e.app_id == *app_id => data.push(e.data.clone()),
```

Hope is still nice.

**Review by itzmeanjan — APPROVED** [review]
Looks good.

**Review by aterentic-ethernal — APPROVED** [review]

**Pull request merged**